OpenEdXMobile/src/main/java/org/edx/mobile/view/EditUserProfileFragment.java

```diff
@@ -498,6 +502,25 @@ private static TextView createField(@NonNull LayoutInflater inflater, @NonNull V
             put("label", field.getLabel());
             put("value", formattedValue);
         }}));
+
+        // set content description when text is truncated. Truncated text can only be retrieved
+        // the view has been laid out. 
+        textView.getViewTreeObserver().addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
+            @Override
+            public void onGlobalLayout() {
+                if (Build.VERSION.SDK_INT < 16) {
+                    textView.getViewTreeObserver().removeGlobalOnLayoutListener(this);
+                } else {
+                    textView.getViewTreeObserver().removeOnGlobalLayoutListener(this);
+                }
+                String contentDescription = TextUtils.ellipsize(textView.getText(),
+                        textView.getPaint(),
+                        (float) textView.getWidth(),
+                        TextUtils.TruncateAt.END).toString();
+                textView.setContentDescription(contentDescription);
```

Comment:
Might it then make sense to set the content description to just the label for this field? This could be done either specifically for just the "bio" field, or generically for any field whose value string exceeds a predefined number of characters. That seems to be a cleaner solution then just truncating TalkBack for only this text view, which may confuse the user.